modules/atmosphere/rendering/renderableatmosphere.h

```diff
@@ -0,0 +1,158 @@
+
+#include <openspace/properties/stringproperty.h>
+#include <openspace/properties/scalar/boolproperty.h>
+#include <openspace/properties/scalar/intproperty.h>
+#include <openspace/properties/scalar/floatproperty.h>
+#include <openspace/util/updatestructures.h>
+
+#include <ghoul/opengl/textureunit.h>
+
+#include <modules/atmosphere/rendering/atmospheredeferredcaster.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace ghoul::opengl {
+    class ProgramObject;
+    class Texture;
+}
+
+namespace openspace {
+
+class AtmosphereDeferredcaster;
+
+struct TransformData;
+
+namespace planetgeometry {
+class PlanetGeometry;
+}
+
+namespace documentation { struct Documentation; }
+namespace planetgeometry { class PlanetGeometry; }
+
+class RenderableAtmosphere : public Renderable {
+public:
+    // Shadow structure
+    typedef struct {
+        std::pair<std::string, float> source;
+        std::pair<std::string, float> caster;
+    } ShadowConf;
+
+    struct ShadowRenderingStruct {
+        float xu, xp;
+        float rs, rc;
+        glm::vec3 sourceCasterVec;
+        glm::vec3 casterPositionVec;
+        bool isShadowing;
+    };
+
+public:
+    RenderableAtmosphere(const ghoul::Dictionary& dictionary);
+
+    bool initialize() override;
+    bool deinitialize() override;
+    bool isReady() const override;
+
+    void render(const RenderData& data, RendererTasks& rendererTask) override;
+    void update(const UpdateData& data) override;
+
+    static documentation::Documentation Documentation();
+
+private: 
+    void computeModelTransformMatrix(const openspace::TransformData & transformData, glm::dmat4 * modelTransform);
+    void updateAtmosphereParameters();
+
+private:
+    
+    AtmosphereDeferredcaster::AtmospherRenderableClass _atmosphereType;
+
+    properties::FloatProperty _atmosphereHeightP;
+    properties::FloatProperty _groundAverageReflectanceP;
+    properties::FloatProperty _rayleighHeightScaleP;
+    properties::FloatProperty _rayleighScatteringCoeffXP;
+    properties::FloatProperty _rayleighScatteringCoeffYP;
+    properties::FloatProperty _rayleighScatteringCoeffZP;
+    properties::BoolProperty  _ozoneEnabledP;
+    properties::FloatProperty _ozoneHeightScaleP;
+    properties::FloatProperty _ozoneCoeffXP;
+    properties::FloatProperty _ozoneCoeffYP;
+    properties::FloatProperty _ozoneCoeffZP;
+    properties::FloatProperty _mieHeightScaleP;
+    properties::FloatProperty _mieScatteringCoeffXP;
+    properties::FloatProperty _mieScatteringCoeffYP;
+    properties::FloatProperty _mieScatteringCoeffZP;
+    properties::FloatProperty _mieScatteringExtinctionPropCoefficientP;
+    properties::FloatProperty _mieAsymmetricFactorGP;
+    properties::FloatProperty _sunIntensityP;
+    properties::FloatProperty _hdrExpositionP;
+    properties::FloatProperty _gammaConstantP;
+    properties::BoolProperty  _sunFollowingCameraEnabledP;
+
+    bool _atmosphereEnabled;
```

Comment:
Some variables have values too small to be used as properties (will be hard for the user handle them), others must be computed based on the values of the properties.